azure-tools-common/src/main/java/com/microsoft/azure/common/function/PackageHandler.java

```diff
@@ -69,20 +69,58 @@
 
 	private static final String FUNCTION_JSON = "function.json";
 	private static final String HOST_JSON = "host.json";
+	public static final String LOCAL_SETTINGS_JSON = "local.settings.json";
 	private static final String EXTENSION_BUNDLE = "extensionBundle";
 
 	private static final BindingEnum[] FUNCTION_WITHOUT_FUNCTION_EXTENSION = { BindingEnum.HttpOutput,
 			BindingEnum.HttpTrigger };
 	private static final String EXTENSION_BUNDLE_ID = "Microsoft.Azure.Functions.ExtensionBundle";
 	private static final String SKIP_INSTALL_EXTENSIONS_BUNDLE = "Extension bundle specified, skip install extension";
 
+	public static final String STAGE_DIR_FOUND = "Azure Function App's staging directory found at: ";
+
 	private IFunctionContext ctx;
 
 	public PackageHandler(IFunctionContext ctx) {
 		this.ctx = ctx;
 	}
 
+
+	private boolean checkStageDirectoryDirty() throws AzureExecutionException {
+		final File stagingDirectory = new File(ctx.getDeploymentStagingDirectoryPath());
+		File jarFileInStaging = new File(stagingDirectory, this.ctx.getProject().getJarArtifact().getFileName().toString());
+		File jarFileInBuild = this.ctx.getProject().getJarArtifact().toFile();
+		if (!jarFileInBuild.exists()) {
+			throw new AzureExecutionException("Artifact file '" + jarFileInBuild + "' cannot be found, please run 'gradle jar' to build the project.");
+		}
+
+		if (jarFileInBuild.length() == 0) {
+			throw new AzureExecutionException("Artifact file '" + jarFileInBuild + "' is empty, please run 'gradle jar' to build the project.");
+		}
+
+		if (!stagingDirectory.exists()) {
+			Log.info("Stage directory not found, build it now...");
+			return true;
+		}
+		if (stagingDirectory.exists() && stagingDirectory.isFile()) {
+			throw new AzureExecutionException("File name conflict: please delete file '" + stagingDirectory.getAbsolutePath() + "', and try again.");
+		}
+		if (jarFileInStaging.exists() && jarFileInStaging.length() == jarFileInBuild.length()) {
+			Log.info(STAGE_DIR_FOUND + ctx.getDeploymentStagingDirectoryPath());
```

Comment:
We should also check whether host.json, function.json exist and properly, and i think it's okay to repackage if user do execute package, or we could add a force build parameter